css-scoping-1/Overview.bs

```diff
@@ -424,6 +424,44 @@ Selecting Slot-Assigned Content: the ''::slotted()'' pseudo-element</h4>
 	The only way to style assigned text nodes
 	is by styling the <a>slot</a> and relying on inheritance.
 
+<h4 id='has-slotted-pseudo'>
+Matching Selectors Against Slot-Assigned Content: the '':has-slotted()'' pseudo-class</h4>
+
+	The <dfn selector>:has-slotted()</dfn> pseudo-class, when evaluated 
+	<a>in the context of a shadow tree</a>,
+	matches the <a>slot</a>, if one of its
+	<a lt="find flattened slottables">assigned elements, after flattening,</a> matches the provided <<compound-selector>>.
```

Comment:
"after flattening" is interesting here because it give "non-slotted" content while being required to gather _all_ slotted content. E.G.:

```
<div>
  <template shadowrootmode="open">
    <style>
      div {
        color: green;
      }
      :has-slotted(div) {
        color: red;
      }
    </style>
    <slot><div>Test</div></slot>
  </template>
</div>
```
In the example above "Test" will display _red_ in the case that flattened elements are leveraged.

However, if you don't use flattened elements:
```
<div>
  <template shadowrootmode="open">
    <div>
      <template shadowrootmode="open">
        <style>
          div {
            color: green;
          }
          :has-slotted(div) {
            color: red;
          }
        </style>
        <slot></slot>
      </template>
      <slot><div>Test</div></slot>
    </div>
  </template>
</div>
```
This "Test" will also be _red_. And, without flattened elements,
```
<div>
  <template shadowrootmode="open">
    <div>
      <template shadowrootmode="open">
        <style>
          div {
            color: green;
          }
          :has-slotted(div) {
            color: red;
          }
        </style>
        <slot></slot>
      </template>
      <div>Test</div>
    </div>
  </template>
</div>
```
This "Test" will also be _red_.

I'm not sure it's a real thing, but is there a place between `::slotted(*)` and `.assignedElements({flatten: true})` that would resolve the _external_ assigned to a slot and not the _default_ ones internal to the shadow host?

Reply:
For consistency, I think it's important that `:has-slotted` matches using the same rules for matchable elements as `::slotted`. 

Concretely, since `::slotted()` never matches either `<slot>` or fallback content, `:has-slotted` also should *not match* when it would by selecting either `<slot>` or fallback content.

See https://github.com/w3c/csswg-drafts/issues/5482. IMO, the spec is still very confusing and possibly indicates that this should match, but the CSS working group resolved that the current behavior is intended in https://github.com/w3c/csswg-drafts/issues/5482.

I think matching against fallback content both for `::slotted` and `:has-slotted` is valuable, but I suggest it should be tackled as a follow-on to this feature.